src/main/java/nextstep/qna/domain/Answer.java

```diff
@@ -9,71 +8,61 @@
 public class Answer {
     private Long id;
 
-    private NsUser writer;
+    private TextBody textBody;
 
     private Question question;
 
-    private String contents;
-
-    private boolean deleted = false;
-
     private LocalDateTime createdDate = LocalDateTime.now();
 
     private LocalDateTime updatedDate;
 
     public Answer() {
     }
 
-    public Answer(NsUser writer, Question question, String contents) {
-        this(null, writer, question, contents);
+    public Answer(TextBody textBody, Question question) {
+        this(null, textBody, question);
     }
 
-    public Answer(Long id, NsUser writer, Question question, String contents) {
+    public Answer(Long id, TextBody textBody, Question question) {
         this.id = id;
-        if(writer == null) {
-            throw new UnAuthorizedException();
-        }
-
-        if(question == null) {
-            throw new NotFoundException();
-        }
-
-        this.writer = writer;
+        textBody.isNull();
+        question.isNull();
         this.question = question;
-        this.contents = contents;
+        this.textBody = textBody;
     }
 
     public Long getId() {
         return id;
     }
 
-    public Answer setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
-    public boolean isDeleted() {
-        return deleted;
+    public Answer create() {
+        textBody = TextBody.of(textBody);
+        return new Answer(id, textBody, question);
     }
 
-    public boolean isOwner(NsUser writer) {
-        return this.writer.equals(writer);
-    }
-
-    public NsUser getWriter() {
-        return writer;
-    }
-
-    public String getContents() {
-        return contents;
+    public TextBody getTextBody() {
+        return textBody;
     }
 
     public void toQuestion(Question question) {
         this.question = question;
     }
 
+    public void compareUser(NsUser loginUser) {
```

Comment:
```suggestion
    public void delete(NsUser loginUser) {
        // 본인이 쓴 답변인지 체크
        this.deleted = true;
    }
```

OOP에서 자신의 상태 값은 상태 값을 가지는 본인이 상태를 변경하도록 구현해 본다.